taworks/taform/views.py

```diff
@@ -73,12 +89,33 @@ def SaveTemp(f):
         tmp.section = line[3]
         tmp.course_name = line[4]
         tmp.instructor_name = line[5]
-        tmp.instructor_email = line[6]
+        tmp.instructor_email = line[6]  
         tmp.save()
 
-def CopyCourses(newtable, oldtable):
+def validate_temp():
+    courses = models.TempCourse.objects.all()
+    for course in courses:
+        if not course.term or course.term<1000 or course.term>9999 or not course.course_id or not course.course_subject or not course.section or not course.course_name:
+            return False
+    return True
+
+def copy_courses(newtable, oldtable):
     models.Course.objects.all().delete()
     queryset = models.TempCourse.objects.all().values('term', 'course_subject', 'course_id', 'section', 'course_name', 'instructor_name', 'instructor_email')
     newobjects = [models.Course(**values) for values in queryset]
     models.Course.objects.bulk_create(newobjects)
 
+def send_file(request):
+    import os, tempfile, zipfile
```

Comment:
from django.core.servers.basehttp import FileWrapper needs to be inside the function or else it throws and error